### ISSUE TITLE
profile photo moderation

### ISSUE THREAD

**Issue by carlos15154** [MissVeronica/um-profile-content-moderation#5]
@missveronica hi why is there no admin approval for profile photos. https://github.com/MissVeronica/um-profile-content-moderation I saw this plugin of yours but it doesn’t include a profile photo. how do we add admin approval when users add a profile photo?

Photos must be moderated because we cannot know what photos each person will upload.

Is there a way to do this? That would be great. Thanks in advance.

**Comment by MissVeronica** [comment]
Yes, I know that Profile and Cover photos should also be included in a Moderation process.

Currently the "Account File Manager" plugin can be used as a tool to display all images by a User from the backend but without a file delete option. Profile and Cover photo removal can be made by an admin logging in at the User profile page.

https://github.com/MissVeronica/um-account-file-manager

**Comment by carlos15154** [comment]
thanks for reply but I know the plugin you mentioned but this is not what I want. When the user changes their profile photo, it should come to me as an email and the profile status should be shown as awaiting approval and the old profile photo should be visible until it is approved. Is it possible to do this?

**Comment by MissVeronica** [comment]
Yes, the "Profile Content Moderation" is made for two different modes.

1.  "Admin Moderation" status after an Profile edit until approved. User profile display is not possible for other Users. An email is sent to Admin about the update. In this mode display of Profile/Cover photo is possible from backend before approval.

2. Delayed Profile update. Saving the Profile update and old profile fields are displayed until approved by an Admin. Email sent to Admin. In this mode photos are not included in the delayed update.

**Comment by carlos15154** [comment]
Yes, I understand this better now. But I want to make it specific only to the profile photo. The plugin covers all the information and only requires admin approval when the profile photo is changed. How can I do this?

**Comment by carlos15154** [comment]
why dont change this option


Moderation Process | Default: User set to UM "Admin Review" status with Profile updated.Option from version 3.6.0: Delayed Profile update until approved by a site Moderator.

**Comment by MissVeronica** [comment]
You can try this code snippet which will set User to Admin review and User is logged out until approved.

```
add_action( 'um_after_move_temporary_files', 'um_after_move_temporary_files_admin_review', 10, 3 );

function um_after_move_temporary_files_admin_review( $user_id, $new_files, $old_files ) {

    UM()->common()->users()->set_as_pending( $user_id, true );
}
```

Install by adding the code snippet to your active theme’s `functions.php` file
or use the “Code Snippets” Plugin

https://wordpress.org/plugins/code-snippets/

**Comment by carlos15154** [comment]
Thanks for your answer, but this code makes the user wait for approval but does not change the new profile photo after approval.

**Comment by carlos15154** [comment]
When I do not accept the profile photo, it does not change. When I say rollback profile, the newly uploaded profile photo needs to be deleted, but this feature did not work.

**Comment by carlos15154** [comment]
When I do not accept the profile photo, it does not change. When I say rollback profile, the newly uploaded profile photo needs to be deleted, but this feature did not work.

**Comment by MissVeronica** [comment]
Right the plugin is not managing anything with Profile/Cover photos.

**Comment by carlos15154** [comment]
Is there a way to do this? How can we create a function to delete the profile photo if we don't approve it?

**Comment by carlos15154** [comment]
> Is there a way to do this? How can we create a function to delete the profile photo if we don't approve it? because photos uploaded without approval may create a security vulnerability.

Is there a way to do this? How can we create a function to delete the profile photo if we don't approve it? because photos uploaded without approval may create a security vulnerability.
